Added spell checking functionality to pre-commit hooks and Makefile.

Prateekshit73 opened the pull request:
<!--
  Thanks for contributing to holidays!
-->

## Proposed change

<!--
  Describe the big picture of your changes.
  Don't forget to link your PR to an existing issue if any.
-->

Added spell checking functionality to pre-commit hooks and Makefile.
Resolves #2308.

## Type of change

<!--
  Type of change you want to introduce. Please, check one (1) box only!
  If your PR requires multiple boxes to be checked, most likely it needs to
  be split into multiple PRs.
-->

- [ ] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/pin/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [x] New feature (new `holidays` functionality in general)

## Checklist

<!--
  Put an `x` in the boxes that apply. You can change them after PR is created.
-->

- [x] I've read and followed the [contributing guidelines](https://github.com/vacanza/holidays/blob/dev/CONTRIBUTING.md).
- [x] I've run `make check` locally; all checks and tests passed.

<!--
  Thanks again for your contribution!
-->

Prateekshit73 committed:
Added spell checking functionality to pre-commit hooks and Makefile.

Prateekshit73 commented:
It still reports 25,729 unknown words. Should I add custom dictionaries like I did for Hindi, or is there another recommended workaround?

arkid15r commented on cspell/cspell.json:
```diff
@@ -0,0 +1,108 @@
+  "enabled": true,
+  "files": ["**/*"],
+  "ignorePaths": [
+    "node_modules/**",
+    ".git/**",
+    ".venv/**",
+    "__pycache__/**",
+    "*.pyc",
+    "dist/**",
+    "build/**",
+    ".mypy_cache/**",
+    ".pytest_cache/**",
+    "docs/_build/**",
+    "snapshots/**",
+    "*.egg-info/**",
+    ".tox/**",
+    "requirements/*.txt",
+    "cspell/dictionaries/**"
```
do we need all of them?

Prateekshit73 commented on cspell/cspell.json:
```diff
@@ -0,0 +1,108 @@
+      "path": "custom-dict.txt"
+    },
+    {
+      "name": "hi",
```
I have already included every available dictionary as a dependency. Here is the complete list for reference:
[Natural Language Dictionaries](https://github.com/streetsidesoftware/cspell-dicts?tab=readme-ov-file#natural-language-dictionaries)

Prateekshit73 committed:
Merge branch 'dev' into comprehensive-spell-checking

Prateekshit73 committed:
Update cspell dependencies and configuration.

Prateekshit73 committed:
Merge branch 'comprehensive-spell-checking' of https://github.com/Prateekshit73/holidays into comprehensive-spell-checking

Prateekshit73 committed:
Merge branch 'dev' into comprehensive-spell-checking

Prateekshit73 committed:
Merge branch 'dev' into comprehensive-spell-checking

Prateekshit73 committed:
Merge branch 'dev' into comprehensive-spell-checking

Signed-off-by: Prateekshit Jaiswal <67361301+Prateekshit73@users.noreply.github.com>

Prateekshit73 committed:
Refine spellcheck command descriptions and enhance error handling in cspell scripts

- Updated usage descriptions in `make.cmd` and `Makefile` to clarify that spellcheck runs on docs and l10n only.
- Improved error handling in `cspell/make.cmd` to notify users of Docker build failures.
- Adjusted `cspell/Makefile` to streamline the spellcheck process.
- Added new entries to `cspell/custom-dict.txt` for additional project-specific terms.
- Enhanced `sort_cspell_dict.py` with better error handling and output messages.

Prateekshit73 committed:
Merge branch 'dev' into comprehensive-spell-checking

Prateekshit73 committed:
Merge branch 'dev' into comprehensive-spell-checking

Prateekshit73 committed:
Merge branch 'dev' into comprehensive-spell-checking

Prateekshit73 committed:
Merge branch 'dev' into comprehensive-spell-checking

Prateekshit73 committed:
Merge branch 'dev' into comprehensive-spell-checking

Prateekshit73 committed:
Add spellcheck workflow to CI/CD and update .gitignore to include node_modules
